src/components/Input/Toggle/Toggle.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import TOGGLE_SIZES from "./constants";
+import {
+  ActiveArea,
+  ReactToggle,
+  ReactToggleTrack,
+  ReactToggleThumb,
+  ReactThumbCenteringContainer
+} from "./Toggle.styles";
+
+export default class Toggle extends Component {
+  handleClick = e => {
+    e.preventDefault();
+    const { onClick, onToggle, value, disabled } = this.props;
+
+    if (disabled) {
+      return;
+    }
+    onClick(e);
```

Comment:
There's a `composeEventHandlers` event handlers utility to handle chaining of internal & user provided event handlers.

Reply:
I am not sure how `composeEventHandler` can help. It executes a function when the function exists. In this case, we don't want to execute any functions when `disable` is `true`.

Reply:
`composeEventHandlers` came in handy after refactoring. Fixed.